pkg/manifests/manifests.go

```diff
@@ -1426,6 +1431,68 @@ func (f *Factory) NewClusterRole(manifest io.Reader) (*rbacv1beta1.ClusterRole,
 	return NewClusterRole(manifest)
 }
 
+// TelemeterClientServiceMonitor generates a new ServiceMonitor for Telemeter client.
+func (f *Factory) TelemeterClientServiceMonitor() (*monv1.ServiceMonitor, error) {
+	sm, err := f.NewServiceMonitor(MustAssetReader(TelemeterClientServiceMonitor))
+	if err != nil {
+		return nil, err
+	}
+
+	sm.Spec.Endpoints[0].TLSConfig.ServerName = fmt.Sprintf("telemeter.%s.svc", f.namespace)
+	sm.Spec.Endpoints[1].TLSConfig.ServerName = fmt.Sprintf("telemeter.%s.svc", f.namespace)
+	sm.Namespace = f.namespace
+
+	return sm, nil
+}
+
+// TelemeterClientDeployment generates a new Deployment for Telemeter client.
+func (f *Factory) TelemeterClientDeployment() (*appsv1.Deployment, error) {
+	d, err := f.NewDeployment(MustAssetReader(TelemeterClientDeployment))
+	if err != nil {
+		return nil, err
+	}
+
+	if f.config.TelemeterClientConfig.BaseImage != "" {
+		image, err := imageFromString(d.Spec.Template.Spec.Containers[2].Image)
+		if err != nil {
+			return nil, err
+		}
+		image.repo = f.config.TelemeterClientConfig.BaseImage
+		image.SetTagIfNotEmpty(f.config.TelemeterClientConfig.Tag)
+		d.Spec.Template.Spec.Containers[0].Image = image.String()
+	}
+
+	d.Namespace = f.namespace
+
+	return d, nil
+}
+
+// TelemeterClientService generates a new Service for Telemeter client.
+func (f *Factory) TelemeterClientService() (*v1.Service, error) {
+	s, err := f.NewService(MustAssetReader(TelemeterClientService))
+	if err != nil {
+		return nil, err
+	}
+
+	s.Namespace = f.namespace
+
+	return s, nil
+}
+
+// TelemeterClientSecret generates a new Secret for Telemeter client.
+func (f *Factory) TelemeterClientSecret() (*v1.Secret, error) {
+	s, err := f.NewSecret(MustAssetReader(TelemeterClientSecret))
+	if err != nil {
+		return nil, err
+	}
+
+	s.StringData["id"] = base64.StdEncoding.EncodeToString([]byte(f.config.TelemeterClientConfig.ClusterID))
+	s.StringData["token"] = base64.StdEncoding.EncodeToString([]byte(f.config.TelemeterClientConfig.PullSecret))
```

Comment:
:+1: this resolves the secret mystery, thanks! :-)

Reply:
I wonder if we should rename this also to `pullSecret` on telemeter side. On second thought, this is fine'ish though, as this token is pretty much opaque to telemeter.

Reply:
If you just set Data to the byte slice of what you want the marshaling will automatically encode it as base64. I generally prefer that as i always forget which base64 encoding is used :)

Reply:
TIL!!!

Reply:
very true, good catch!